_im-plugin/append-only-index.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: default
+title: Append-only Index
+nav_order: 14
+redirect_from:
+  - /opensearch/append-only-index/
+---
+
+# Append only index
+
+An append-only index is an immutable index that only allows document ingestion (appending) while blocking all forms of updates or deletions after initial document creation.
+When you enable the append-only setting for an index, OpenSearch prevents any modifications to existing documents. You can only add new documents to the index.
+
+When an index is configured as append-only, the following operations will return an error:
+- Document update call (Update API)
+- Document delete call (Delete API)
+- Update by query call
+- Delete by query call
+- Bulk API call that include update, delete, or upsert actions
+- Bulk API call with an index action with a custom document ID.
+
+Append-only indices offer several advantages:
+- Optimised performance as it eliminates costly updates/delete operations.
+- Optimised storage and segment merges as it eliminate soft deletes or version tracking.
+- Enables future optimisations like auto-rollovers and efficient warm tiering.
+
+Ideal use cases of append only indices are immutable workloads like logs, metrics, observability or security event data where we do not modify data once ingested.
+
+
+## Example usage
+
+The following example request creates a new index named `my-append-only-index` which has all forms of updates disabled on it:
+
+```json
+PUT /my-append-only-index
```

Comment:
Should this be through the `cluster-settings` API?

Reply:
This is an index level setting.